### PR TITLE
feat: add capability to plot file size chronology for valid files

### PR DESCRIPTION
**Pull request by slawrence87544**
body: this commit will make it possible to plot observation file
size chronology for all valid files only.

### CODE REVIEW

File: src/obs_inv_utils/plot_generator.py [hunk 1]
```diff
@@ -19,4 +43,147 @@ def __post_init__(self):
 
     def plot_timeline(self):
         data = oiq.get_filesize_timeline_data(self.min_instances)
-        print(f'Observation inventory file filesize data: {data}')
+
+        unique_rows = data.sort_values(
```
Comment (frolovsa) [inline src/obs_inv_utils/plot_generator.py]: maybe add a comment on why there are duplicates etc.

File: src/obs_inv_utils/plot_generator.py [hunk 2]
```diff
@@ -19,4 +43,147 @@ def __post_init__(self):
...
+        ).drop_duplicates(
+            subset=[
+                'filename',
+                'obs_day'
+            ],
+            keep='last'
+        )
+        data_index = unique_rows.index
+
+        uf = data.sort_values(
```
Comment (frolovsa) [inline src/obs_inv_utils/plot_generator.py:53]: maybe add a comment on why there are duplicates and how you filter for them

File: src/obs_inv_utils/plot_generator.py [hunk 3]
```diff
@@ -19,4 +43,147 @@ def __post_init__(self):
...
+            'inserted_at'
+        ).drop_duplicates(
+            ['filename', 'obs_day'],
+            keep='last'
+        ).dropna(subset=['cycle_time'])
+
+        # bad = uf.loc[uf['cycle_time'] == np.nan]
+        # print(f'bad: {bad}')
+
+        uf['generic_fn'] = uf['prefix'] + '.tag.' + uf['un']
```
Comment (frolovsa) [inline src/obs_inv_utils/plot_generator.py:70]: add a comment on why you doing this massaging. E.g. 
`# make field names used in plotting`

**Review by frolovsa — CHANGES_REQUESTED** [review]
Steve. looks good. thank you. 
General comment. It would be great to see some additional comments on why and how you are massaging the data before plotting.

**Comment by HenryRWinterbottom** [comment]
Looks good to me.

**Review by HenryRWinterbottom — APPROVED** [review]

File: src/obs_inv_utils/plot_generator.py (hunk 2 above)
Comment (slawrence87544) [reply src/obs_inv_utils/plot_generator.py:53]: Added clarifying comments.

File: src/obs_inv_utils/plot_generator.py (hunk 1 above)
Comment (slawrence87544) [reply src/obs_inv_utils/plot_generator.py]: Removed code section as it was unnecessary.

**Review by frolovsa — APPROVED** [review]

**Review by HenryRWinterbottom — COMMENTED** [review]
These look good to me.

**Pull request merged**